python_src/src/lamp_py/ingestion/pipeline.py

```diff
@@ -71,23 +83,67 @@ def ingest(metadata_queue: Queue) -> None:
         file_prefix=DEFAULT_S3_PREFIX,
     )
 
-    ingest_files(files, metadata_queue)
+    processed_files = ingest_files(files, metadata_queue)
+
+    process_logger.log_complete()
+
+    return processed_files
+
+
+def wait_for_deletion(filenames: List[str]) -> None:
+    """
+    wait for all of the files in the filenames list to be removed from the
+    incoming bucket. log how many files are still
+    """
+    process_logger = ProcessLogger(
+        "wait_for_deletion", file_count=len(filenames)
+    )
+    process_logger.log_start()
+
+    files_to_delete = set(filenames)
+    attempts = 0
+    while len(files_to_delete) > 0 and attempts < 5:
+        existing_files = set(
+            file_list_from_s3(
+                bucket_name=os.environ["INCOMING_BUCKET"],
+                file_prefix=DEFAULT_S3_PREFIX,
+            )
+        )
+
+        files_to_delete &= existing_files
+        if len(files_to_delete) == 0:
+            break
+
+        # increment attempts and wait at most 30 seconds to check again
+        attempts += 1
+        time.sleep(min(30, int(len(files_to_delete) / 100)))
```

Comment:
This random sleep feels like were crossing over into dictating behavior instead of observing it.